### DISCUSSION TITLE
hvac package is still redirecting to the v1 version

### DISCUSSION THREAD

**Discussion by surajtikoo** [hvac/hvac#973] [Q&A]
Hi Team,
I am using the hvac package with my python code to read the secrets however the secrets are placed under the v2 version.
I am using the 1.1.0 version for hvac. When i am truing to read the secrets the system is redirecting to the V1 irrespective of giving the V2 path.

         path = 'kv-test/test'
        response = client.secrets.kv.v2.read_secret(path=path)

The system is giving the error
"errorMessage": "1 error occurred:\n\t* permission denied\n\n, on get https://xxxxxxxx/v1/secret/data/kv-test/test",

The call is still going to the V1 version which is a bit strange to me.

**Comment by briantist** [comment]
Hi @surajtikoo ! The `/v1/` in the API path refers to the major version of the Vault API, not the secret engine, and only 1 version exists right now, so every API call will look like that.

I don't have visibility into your system, but if I had t guess I might say that in your path `kv-test/test`, you might be trying to read the key `test` within the key/value pairs stored in `kv-test`.

If that's the case, it won't work that way, you must retrieve the `kv-test` secret (path), and it will return the whole dictionary of key/value pairs, then you read what you need from the returned dictionary.

**Comment by dshvedchenko** [comment]
@surajtikoo if  'kv-test/test'  is mount_point/secret , then you need to read it as

client.secrets.kv.v2.read_secret(path='test', mount_point='kv_test')

if you omit mount_point, then it will try to read from default _mount point , that can be V1 ...
